src/avtt/content-script.js

```diff
@@ -0,0 +1,580 @@
+        const [_, ...rows] = [...match.matchAll(matchRow)].map(([_1, _2, firstRow, secondRow]) => {
+            return `| ${firstRow} | ${secondRow} |`
+        });
+        return `
+
+| ${firstHead} | ${secondHead} |
+| :--- | :--- |
+${rows.join('\n')}
+
+`;
+    })
+}
+
+function formatNotesInDescription(description) {
+    return description.replace(/Notes: [^\n]*/g, (match) => `_${match}_`);
+}
+
+function formatBoldedSectionInDescription(description) {
+    return description.replace(/([^\n\.])*\.\&nbsp\;/g, match => `\n**${match}**`)
+}
+
+function formatBulletListsInDescription(description) {
+    return description.replace(/^(?!\|)(([^\n\r]+)\n)+([^\n\r]+)/gm, match => {
+        return match.split('\n').map(r => `* ${r}`).join('\n');
+    })
+}
+
+function formatSeparateParagraphsInDescription(description) {
+    return description.replace(/^(?!\|)(([^\n\r]+)\n)+([^\n\r]+)/gm, match => {
+        return match.split('\n').join('\n\n');
+    })
+}
+
+
+
+function advantageRoll(request, name, roll) {
+    return {
+        [RollType.NORMAL]: [[name, roll]],
+        [RollType.DOUBLE]: [[name, roll], ["2nd time", roll]],
+        [RollType.THRICE]: [[name, roll], ["2nd time", roll], ["3rd time", roll]],
+        [RollType.ADVANTAGE]: [[name, roll], ["Advantage", roll]],
+        [RollType.DISADVANTAGE]: [[name, roll], ["Disadvantage", roll]],
+        [RollType.SUPER_ADVANTAGE]: [[name, roll], ["Advantage", roll], ["Super Advantage", roll]],
+        [RollType.SUPER_DISADVANTAGE]: [[name, roll], ["Disadvantage", roll], ["Super Disadvantage", roll]],
+    }[request.advantage] || [[name, roll]]
+}
+
+function rollSkill(request, custom_roll_dice = "") {
+    let modifier = request.modifier;
+    // Custom skill;
+    if (request.ability === "--" && request.character.abilities.length > 0) {
+        let prof = "";
+        let prof_val = "";
+        let reliableTalent = false;
+        if (request.proficiency === "Proficiency") {
+            prof = "PROF";
+            prof_val += request.character.proficiency;
+            reliableTalent = request.reliableTalent;
+        } else if (request.proficiency === "Half Proficiency") {
+            prof = "HALF-PROFICIENCY";
+            prof_val += "floor(" + request.character.proficiency + " / 2)";
+        } else if (request.proficiency === "Expertise") {
+            prof = "EXPERTISE";
+            prof_val += `${request.character.proficiency} * 2`;
+            reliableTalent = request.reliableTalent;
+        }
+        const d20 = reliableTalent ? "1d20min10" : "1d20";
+        return {
+            icon: "bunker-soldier-standing",
+            color: "#7ED321",
+            title: request.skill + " Skill Check",
+            message: template([
+                ...advantageRoll(request, "Skill Check", `!(${d20}${prof_val.length > 0 ? " + " + prof_val : ""}${custom_roll_dice.length > 0 ? " + " + custom_roll_dice : ""})`)
+            ])
```

Comment:
why spread into a list here? I would also suggest defining the roll in a separate variable so it's not one huge line.

```js
          const roll = `!(${d20}${prof_val.length > 0 ? " + " + prof_val : ""}${custom_roll_dice.length > 0 ? " + " + custom_roll_dice : ""})`;
 ....
            message: template(advantageRoll(request, "Skill Check", roll))
```

Reply:
The spread here I think may be a leftover from usually expecting after the advantage roll to add damage or other rows. Changed it to the recomandation.